explorer/e2e/anvil/anvil-filters.spec.ts

```diff
@@ -0,0 +1,168 @@
+  page,
+}) => {
+  await testFilterPresence(page, anvilTabs.donors, anvilFilters);
+});
+
+test("Check that all filters exist on the BioSamples tab and are clickable", async ({
+  page,
+}) => {
+  await testFilterPresence(page, anvilTabs.biosamples, anvilFilters);
+});
+
+test("Check that all filters exist on the Activities tab and are clickable", async ({
+  page,
+}) => {
+  await testFilterPresence(page, anvilTabs.activities, anvilFilters);
+});
+
+test("Check that all filters exist on the Files tab and are clickable", async ({
+  page,
+}) => {
+  await testFilterPresence(page, anvilTabs.files, anvilFilters);
+});
+
+test("Check that the first filter on the Datasets tab creates at least one checkbox, and that checking up to the first five does not cause an error and does not cause there to be no entries in the table", async ({
+  page,
+}) => {
+  test.setTimeout(120000);
+  // Goto the datasets tab
+  await page.goto(anvilTabs.datasets.url);
+  await expect(
+    page.getByRole("tab").getByText(anvilTabs.datasets.tabName)
+  ).toBeVisible();
+
+  // Select a filter
+  await page
+    .getByRole("button")
+    .getByText(
+      filterRegex(anvilFilters[Math.floor(Math.random() * anvilFilters.length)])
+    )
+    .click();
+  // Expect all checkboxes to be unchecked initially and to work properly
+  await expect(page.getByRole("checkbox").first()).toBeVisible();
+  const allCheckboxes = await page.getByRole("checkbox").all();
+  for (let i = 0; i < allCheckboxes.length && i < 5; i++) {
+    const checkbox = allCheckboxes[i];
+    await checkbox.scrollIntoViewIfNeeded();
+    await expect(checkbox).not.toBeChecked();
+    await checkbox.click();
+    await expect(checkbox).toBeChecked();
+  }
+  await page.locator("body").click();
+  await expect(getFirstElementTextLocator(page, 0)).toBeVisible();
+});
+
+test("Check that filter checkboxes are persistent across pages on an arbitrary filter", async ({
+  page,
+}) => {
+  test.setTimeout(120000);
+  await testFilterPersistence(
+    page,
+    anvilFilters[FILE_FORMAT_INDEX],
+    anvilTabTestOrder.map((x) => anvilTabs[x])
+  );
+});
+
+test("Check that filter menu counts match actual counts on the Datasets tab", async ({
+  page,
+}) => {
+  test.setTimeout(120000);
+  const result = await testFilterCounts(
+    page,
+    anvilTabs.datasets,
+    FILTER_INDEX_LIST.map((x) => anvilFilters[x]),
+    anvilTabs.datasets.maxPages ?? 0
+  );
+  if (!result) {
+    test.fail();
+  }
+});
+
+test("Check that filter menu counts match actual counts on the Activities tab", async ({
+  page,
+}) => {
+  test.setTimeout(120000);
+  await testFilterCounts(
+    page,
+    anvilTabs.activities,
+    FILTER_INDEX_LIST.map((x) => anvilFilters[x]),
+    anvilTabs.activities.maxPages ?? 0
+  );
+});
+
+test("Check that the blue filter bubbles match the selected filter for an arbitrary filter on the Files tab", async ({
+  page,
+}) => {
+  test.setTimeout(120000);
+  await testFilterBubbles(
```

Comment:
It looks like we call the bubbles "tags"; can you rename bubbles to "tags" here and throughout?